astropy/coordinates/representation.py

```diff
@@ -1564,6 +1913,23 @@ def _check_base(cls, base):
                             '{0}, not {1}.'.format(cls.base_representation,
                                                    type(base)))
 
+    def _get_deriv_key(self, base):
+        """Given a base (representation instance), determine the unit of the
+        derivative by removing the representation unit from the component units
+        of this differential.
+        """
+
+        # TODO: better validation by checking, e.g., _compatible_differentials
+        # for this class
+        self._check_base(base)
+
+        for name in base.components:
+            comp = getattr(base, name)
+            d_comp = getattr(self, 'd_{0}'.format(name), None)
+            if d_comp:
+                d_unit = comp.si.unit / d_comp.si.unit
+                return str(d_unit)
```

Comment:
I think this is too inprecise:
```
str(u.deg.si/(u.mas/u.yr).si)
# '1.13607e+14 s'
```
I think we just want the SI base and power. The easiest is a bit of a trick, but it works:
```
str(u.Quantity(1., comp.unit / d_comp.unit).si.unit)
str(u.Quantity(1., u.deg/(u.mas/u.yr**2)).si.unit)
# 's2'
```